flow360/component/simulation/models/solver_numerics.py

```diff
@@ -0,0 +1,363 @@
+    order_of_accuracy: Literal[1, 2] = pd.Field(2)
+    equation_evaluation_frequency: PositiveInt = pd.Field(1)
+    linear_solver: LinearSolver = pd.Field(LinearSolver())
+
+
+class NavierStokesSolver(GenericSolverSettings):
+    """:class:`NavierStokesSolver` class for setting up compressible Navier-Stokes solver
+
+    Parameters
+    ----------
+
+    absolute_tolerance :
+        Tolerance for the NS residual, below which the solver goes to the next physical step
+
+    relative_tolerance :
+        Tolerance to the relative residual, below which the solver goes to the next physical step. Relative residual is
+        defined as the ratio of the current pseudoStep’s residual to the maximum residual present in the first
+        10 pseudoSteps within the current physicalStep. NOTE: relativeTolerance is ignored in steady simulations and
+        only absoluteTolerance is used as the convergence criterion
+
+    CFL_multiplier :
+        Factor to the CFL definitions defined in “timeStepping” section
+
+    kappa_MUSCL :
+        Kappa for the MUSCL scheme, range from [-1, 1], with 1 being unstable. The default value of -1 leads to a 2nd
+        order upwind scheme and is the most stable. A value of 0.33 leads to a blended upwind/central scheme and is
+        recommended for low subsonic flows leading to reduced dissipation
+
+    update_jacobian_frequency :
+        Frequency at which the jacobian is updated.
+
+    equation_evaluation_frequency :
+        Frequency at which to update the compressible NS equation in loosely-coupled simulations
+
+    max_force_jac_update_physical_steps :
+        When which physical steps, the jacobian matrix is updated every pseudo step
+
+    order_of_accuracy :
+        Order of accuracy in space
+
+    limit_velocity :
+        Limiter for velocity
+
+    limit_pressure_density :
+        Limiter for pressure and density
+
+    numerical_dissipation_factor :
+        A factor in the range [0.01, 1.0] which exponentially reduces the dissipation of the numerical flux.
+        The recommended starting value for most low-dissipation runs is 0.2
+
+    linear_solver:
+        Linear solver settings
+
+    low_mach_preconditioner:
+        Uses preconditioning for accelerating low Mach number flows.
+
+    low_mach_preconditioner_threshold:
+        For flow regions with Mach numbers smaller than threshold, the input Mach number to the preconditioner is
+        assumed to be the threshold value if it is smaller than the threshold.
+        The default value for the threshold is the freestream Mach number.
+
+    Returns
+    -------
+    :class:`NavierStokesSolver`
+        An instance of the component class NavierStokesSolver.
+
+    Example
+    -------
+    >>> ns = NavierStokesSolver(absolute_tolerance=1e-10)
+    """
+
+    absolute_tolerance: PositiveFloat = pd.Field(1.0e-10)
+
+    CFL_multiplier: PositiveFloat = pd.Field(1.0)
+    kappa_MUSCL: pd.confloat(ge=-1, le=1) = pd.Field(-1)
```

Comment:
Let's collect feedback on renaming stuff during internal testing and do all at once later. Otherwise it is a front end nightmare if we constantly changes things here and there.